src/commands/postDeploy/getLinuxDetectorError.ts

```diff
@@ -3,83 +3,97 @@
 
-    const datasets: detectorDataset[] = responseJson.properties.dataset;
-    const vsCodeIntegration: string = 'Latest time seen by detector. To be used in VSCode integration.';
-    const timestampTable: detectorTable | undefined = findTableByRowValue(datasets, vsCodeIntegration);
+    const insightLogTable: detectorTable | undefined = findTableByName(responseJson.properties.dataset, 'insight/logs');
 
-    // if we can't find the timestamp, exit and try again
-    if (!timestampTable) {
+    if (!insightLogTable) {
         return undefined;
     }
 
-    const timestamp: Date = new Date(getValuesByColumnName(context, timestampTable, ColumnName.value));
-    if (validateTimestamp(context, timestamp, deployResultTime)) {
-
-        const criticalErrorTable: detectorTable | undefined = findTableByRowValue(datasets, 'Critical');
-        const failureCountTable: detectorTable | undefined = findTableByColumnName(datasets, ColumnName.failureCount);
-
-        // if the criticalError table or failureCounts aren't defined, we can't display anything to the user
-        if (!criticalErrorTable || !failureCountTable) {
+    const rawApplicationLog: string = getValuesByColumnName(context, insightLogTable, ColumnName.value);
+    const insightDataset: detectorDataset[] = <detectorDataset[]>JSON.parse(rawApplicationLog);
+
+    let insightTable: detectorTable;
+    let detectorTimestamp: string;
+
+    // [1] 2020-04-21T18:23:50 is the format of the timestamp in the insight response, this RegExp will remove the brackets
+    const bracketsAndSpace: RegExp = /\[.*?\]\s/;
+    const appInsightTable: detectorTable | undefined = findTableByName(insightDataset, 'application/insight');
+    if (appInsightTable) {
+        insightTable = appInsightTable;
+        context.telemetry.properties.insight = 'app';
+        detectorTimestamp = getValuesByColumnName(context, appInsightTable, ColumnName.dataName).replace(bracketsAndSpace, '');
+    } else {
+        // if there are no app insights, defer to the Docker container
+        const dockerInsightTable: detectorTable | undefined = findTableByName(insightDataset, 'docker/insight');
+        if (!dockerInsightTable) {
             return undefined;
         }
 
-        const errorMessages: string | undefined = getValuesByColumnName(context, criticalErrorTable, ColumnName.message);
-        const failureCount: string | undefined = getValuesByColumnName(context, failureCountTable, ColumnName.failureCount);
+        insightTable = dockerInsightTable;
+        context.telemetry.properties.insight = 'docker';
+        detectorTimestamp = getValuesByColumnName(context, dockerInsightTable, ColumnName.dataName).replace(bracketsAndSpace, '');
+    }
 
-        context.telemetry.properties.errorMessages = JSON.stringify(errorMessages);
-        context.telemetry.properties.failureCount = failureCount;
+    detectorTimestamp = moment.utc(detectorTimestamp).format(timeFormat);
```

Comment:
Rather than doing `.replace(bracketsAndSpace, '');` twice above, just do it here.